kernel/dev/serial/rs232.cpp

```diff
@@ -101,3 +101,44 @@ px_ring_buff_t* px_rs232_init_buffer(int size) {
         return NULL;
     }
 }
+
+px_ring_buff_t* px_rs232_get_buffer() {
+    // Can return NULL. This is documented.
+    return read_buffer;
+}
+
+char px_rs232_get_char() {
+    // Grab the last byte and convert to a char
+    uint8_t data;
+    px_ring_buffer_dequeue(read_buffer, &data);
+    return (char)data;
+}
+
+char* px_rs232_get_str() {
+    int idx = 0;
+    char* str = (char*)malloc(sizeof(char) * 1024);
+    // Keep reading until the buffer is empty or
+    // a newline is read.
+    while (!px_ring_buffer_is_empty(read_buffer)) {
+        uint8_t byte;
+        px_ring_buffer_dequeue(read_buffer, &byte);
+        str[idx] = (char)byte;
+        ++idx;
+        // Break if it's a newline
+        if ((char)byte == '\n') {
+            break;
+        }
+    }
+    // Add the null terminator
+    str[idx] = '\0';
```

Comment:
If the buffer was full, then this will write past the end of the array.

Reply:
Resolved in the latest commit (hopefully)